src/main/java/com/carpool/car_pool/controllers/RideOfferController.java

```diff
@@ -36,19 +36,28 @@ public ResponseEntity<Long> createRideOffer(
         return ResponseEntity.ok(rideOfferService.createRideOffer(rideOfferRequest, currentUser));
     }
 
-    @GetMapping("/details")
+    @GetMapping("/details/{id}")
     public ResponseEntity<RideOfferResponse> viewRideOfferDetails(
-            @RequestParam @Valid Long ID
-            ) {
-        return ResponseEntity.ok(rideOfferService.viewRideOfferDetail(ID));
+            @PathVariable Long id
+    ) {
+        return ResponseEntity.ok(rideOfferService.viewRideOfferDetail(id));
     }
 
-    @PutMapping("/details")
+    @PutMapping("/details/{id}")
     public ResponseEntity<RideOfferResponse>  editRideOfferDetails(
-            @RequestBody EditRideOfferRequest editRideofferRequest
+            @RequestBody EditRideOfferRequest editRideOfferRequest,
+            @PathVariable String id // TODO: does this make sense? should I check if id == editRideOfferRequest.id?
```

Comment:
Can delete this there is already a id variable in the EdieRideOfferRequest. Also all id's are long for easier autogeneration